### DISCUSSION TITLE
Is it possible to add a response header to the openapi spec that is generated?

### DISCUSSION THREAD

**Discussion by jceresini** [fastapi/fastapi#7987] [Questions] [answered]
**Description**

How can I [...]?
Add a response header to the openapi spec that is generated.

**Additional context**
I was hoping to use the response headers for pagination, something similar to the following:
```
@api.get('/things')
def list_things(response: Response, params: dict = Depends(pagination_params)):

    things_resp = get_things(**params)
    if things_resp.next_page_token:
        response.headers['next_page_token'] = things_resp.next_page_token

    return things_resp.things
```

But I'd like to inform users of my API that they may see a `next_page_token` header in the response by way of the OpenAPI spec. I read through a bunch of the documentation but didn't see a solution for this use case

**Comment by jceresini** [comment, marked as answer]
Sorry, just found my answer here: https://github.com/tiangolo/fastapi/issues/132#issuecomment-480328266

I'll close this question

**Reply by YuriiMotov** [reply]
I'll just add code example:
```python
from fastapi import FastAPI

app = FastAPI()

@app.get(
    "/",
    responses={
        200: {
            "headers": {
                "Location": {
                    "description": "Location header",
                    "schema": {"type": "string"},
                }
            }
        }
    },
)
def read_root():
    return ""

```

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue :+1:
